sechub-scan/src/test/java/com/mercedesbenz/sechub/domain/scan/asset/AssetServiceTest.java

```diff
@@ -252,6 +254,34 @@ void ensureAssetFileInStorageAvailableAndHasSameChecksumAsInDatabase_in_storage_
 
     }
 
+    @Test
+    void delete_non_existing_single_asset_file_throws_exception() throws IOException {
+        /* prepare */
+        String assetId = "asset1";
+        String fileName = "file1.txt";
+
+        AssetStorage assetStorage = mock();
+        when(storageService.createAssetStorage(assetId)).thenReturn(assetStorage);
+        doNothing().when(assetStorage).delete(fileName);
+
+        /* execute + test */
+        assertThatThrownBy(() -> serviceToTest.deleteAssetFile(assetId, fileName)).isInstanceOf(NotFoundException.class).hasMessageContaining(assetId)
+                .hasMessageContaining(fileName);
+    }
+
+    @Test
+    void delete_non_existing_asset_throws_exception() throws IOException {
+        /* prepare */
+        String assetId = "asset1";
+
+        AssetStorage assetStorage = mock();
+        when(storageService.createAssetStorage(assetId)).thenReturn(assetStorage);
+        doNothing().when(assetStorage).deleteAll();
+
+        /* execute + test */
+        assertThatThrownBy(() -> serviceToTest.deleteAsset(assetId)).isInstanceOf(NotFoundException.class).hasMessageContaining(assetId);
+    }
+
     private AssetFile createAssetFileMock(String fileName, String checksum) {
         AssetFileCompositeKey mockKey1 = mock(AssetFileCompositeKey.class);
         when(mockKey1.getFileName()).thenReturn(fileName);
```

Comment:
Not the right place, but change existing Integration test

- AssetScenario1IntTest

and execute there the delete service calls (after delete is already done) + check via `TestAPi.expectHttpFailure(...)`

I think at the end of the existing integration tests should be a good place for this.